src/trpc/router/groupRouter.ts

```diff
@@ -21,33 +22,41 @@ export const groupRouter = router({
     }),
 
     edit: protectedProcedure.input(editGroupSchema).mutation(async ({ input, ctx }) => {
+        const group = await groupMethods.getById(input.groupId);
         accessCheckAnyOf(
             checkRoleForAccess(ctx.session.user.role, 'editFullGroupTree'),
             checkRoleForAccess(ctx.session.user.role, 'editAdministratedGroupTree'),
+            await groupAccess.isEditable(ctx.session.user, group.id),
         );
         return groupMethods.edit(input);
     }),
 
-    archive: protectedProcedure.input(z.string()).mutation(({ input, ctx }) => {
+    archive: protectedProcedure.input(z.string()).mutation(async ({ input, ctx }) => {
+        const group = await groupMethods.getById(input);
         accessCheckAnyOf(
             checkRoleForAccess(ctx.session.user.role, 'editFullGroupTree'),
             checkRoleForAccess(ctx.session.user.role, 'editAdministratedGroupTree'),
+            await groupAccess.isEditable(ctx.session.user, group.id),
```

Comment:
you await getting a group by id only to use id here? why wouldn't you just pass id?